environment/src/main/java/com/sequenceiq/environment/proxy/service/ProxyConfigService.java

```diff
@@ -0,0 +1,91 @@
+
+import com.google.common.collect.Sets;
+import com.sequenceiq.cloudbreak.auth.altus.Crn;
+import com.sequenceiq.cloudbreak.common.exception.NotFoundException;
+import com.sequenceiq.cloudbreak.logger.MDCBuilder;
+import com.sequenceiq.environment.proxy.domain.ProxyConfig;
+import com.sequenceiq.environment.proxy.repository.ProxyConfigRepository;
+
+@Service
+public class ProxyConfigService  {
+
+    private final ProxyConfigRepository proxyConfigRepository;
+
+    public ProxyConfigService(ProxyConfigRepository proxyConfigRepository) {
+        this.proxyConfigRepository = proxyConfigRepository;
+    }
+
+    public ProxyConfig get(Long id) {
+        return proxyConfigRepository.findById(id).orElseThrow(notFound("Proxy configuration", id));
+    }
+
+    public ProxyConfig deleteInAccount(String name, String accountId) {
+        ProxyConfig proxyConfig = proxyConfigRepository.findByNameInAccountId(name, accountId)
+                .orElseThrow(notFound("Proxy config with name:", name));
+        MDCBuilder.buildMdcContext(proxyConfig);
+        proxyConfigRepository.delete(proxyConfig);
+        return proxyConfig;
+    }
+
+    public Set<ProxyConfig> deleteMultipleInAccount(Set<String> names, String accountId) {
+        Set<ProxyConfig> toBeDeleted = getByNamesForAccountId(names, accountId);
+        return toBeDeleted.stream()
+                .map(credential -> deleteInAccount(credential.getName(), accountId))
+                .collect(Collectors.toSet());
+    }
+
+    private Set<ProxyConfig> getByNamesForAccountId(Set<String> names, String accountId) {
+        Set<ProxyConfig> results = proxyConfigRepository.findByNamesInAccount(names, accountId);
+        Set<String> notFound = Sets.difference(names,
+                results.stream().map(p -> p.getName()).collect(Collectors.toSet()));
+
+        if (!notFound.isEmpty()) {
+            throw new NotFoundException(String.format("No proxy config found with name(s) '%s'",
+                    notFound.stream().map(name -> '\'' + name + '\'').collect(Collectors.joining(", "))));
+        }
+
+        return results;
+    }
+
+    public ProxyConfig getByNameForAccountId(String name, String accountId) {
+        return proxyConfigRepository.findByNameInAccountId(name, accountId)
+                .orElseThrow(notFound("No proxy config found with name", name));
+    }
+
+    public Set<ProxyConfig> listInAccount(String accountId) {
+        return proxyConfigRepository.findAllInAccount(accountId);
+    }
+
+    public ProxyConfig create(ProxyConfig proxyConfig, String accountId) {
+        proxyConfig.setResourceCrn(createCRN(accountId));
+        proxyConfig.setAccountId(accountId);
+        try {
+            return proxyConfigRepository.save(proxyConfig);
+        } catch (DataIntegrityViolationException e) {
+            throw new AccessDeniedException("Access denied");
+        }
+    }
+
+    private String createCRN(@Nonnull String accountId) {
```

Comment:
in my opinion, checking null on accountId is not necessary when it's coming from inside the class. If this value/field is coming outside from this class and it must not be null, then that method should check it which gets it in the first hand.

Reply:
fine